components/google-cloud/google_cloud_pipeline_components/experimental/remote/gcp_launcher/batch_prediction_job_remote_runner.py

```diff
@@ -0,0 +1,140 @@
+    gca_job_state.JobState.JOB_STATE_CANCELLED,
+    gca_job_state.JobState.JOB_STATE_PAUSED,
+)
+
+
+def create_batch_prediction_job(type, project, location, payload, gcp_resources):
+  """Create and poll batch prediction job status till it reaches a final state.
+
+  This follows the typical launching logic 1.
+
+  Read if the batch prediction job already exists in gcp_resources
+     - If already exists, jump to step 3 and poll the job status. This happens
+     if the
+       launcher container experienced unexpected termination, such as preemption
+  2. Deserialize the payload into the job spec and create the batch prediction
+  job.
+  3. Poll the batch prediction job status every _POLLING_INTERVAL_IN_SECONDS
+  seconds
+     - If the batch prediction job is succeeded, return succeeded
+     - If the batch prediction job is cancelled/paused, it's an unexpected
+     scenario so return failed
+     - If the batch prediction job is running, continue polling the status
+  Also retry on ConnectionError up to _CONNECTION_ERROR_RETRY_LIMIT times during
+  the poll.
+  """
+  client_options = {'api_endpoint': location + '-aiplatform.googleapis.com'}
+  client_info = gapic_v1.client_info.ClientInfo(
+      user_agent='google-cloud-pipeline-components',)
+
+  batch_prediction_job_uri_prefix = f"https://{client_options['api_endpoint']}/v1/"
+
+  # Initialize client that will be used to create and send requests.
+  job_client = aiplatform.gapic.JobServiceClient(
+      client_options=client_options, client_info=client_info)
+
+  # Instantiate GCPResources Proto
+  batch_prediction_job_resources = GcpResources()
+  batch_prediction_job_resource = batch_prediction_job_resources.resources.add()
+
+  # Check if the Custom job already exists
+  if path.exists(gcp_resources) and os.stat(gcp_resources).st_size != 0:
+    with open(gcp_resources) as f:
+      serialized_gcp_resources = f.read()
+      batch_prediction_job_resource = json_format.Parse(
+          serialized_gcp_resources, batch_prediction_job_resource)
+      batch_prediction_job_name = batch_prediction_job_resource.resource_uri[
+          len(batch_prediction_job_uri_prefix):]
+
+      logging.info(
+          'BatchPredictionJob name already exists: %s. Continue polling the status',
+          batch_prediction_job_name)
+  else:
+    parent = f'projects/{project}/locations/{location}'
+    job_spec = json.loads(payload, strict=False)
+    create_batch_prediction_job_response = job_client.create_batch_prediction_job(
+        parent=parent, batch_prediction_job=job_spec)
+    batch_prediction_job_name = create_batch_prediction_job_response.name
+
+    # Write the job proto to output
+    batch_prediction_job_resource.resource_type = 'BachPredictionJob'
+    batch_prediction_job_resource.resource_uri = f'{batch_prediction_job_uri_prefix}{batch_prediction_job_name}'
+
+    with open(gcp_resources, 'w') as f:
+      f.write(json_format.MessageToJson(batch_prediction_job_resource))
```

Comment:
the batch prediction is a little bit special in that the component also outputs the job as an artifact
https://github.com/kubeflow/pipelines/blob/master/components/google-cloud/google_cloud_pipeline_components/aiplatform/batch_predict_job/component.yaml#L172

For backward compatiblity, you probably need to outputs that as well.